### DISCUSSION TITLE
How to add template tag for label template?

### DISCUSSION THREAD

**Discussion by adamwolf** [inventree/InvenTree#6564] [General]
Hiya folks!

I want to do some things in a new label template that seem to need a new template tag.  What's a good way to do this?  Can I add the tag via a plugin, and then use it in the label template file?

Thanks!

**Comment by matmair** [comment]
You can register template tags following normal Django patterns if the code is included in a plugin with AppMixin.

**Comment by wolflu05** [comment]
Otherwise if you don't want to use an appmixin, you can also use the report mixin and provide a custom [context](https://docs.inventree.org/en/stable/extend/plugins/report/#add-label-context).
